drivers: i2c: i2c_nrfx_twim: add DMM usage in driver

mstasiaknordic opened the pull request:
Added DMM buffer preparation to I2C TWIM driver.
Adds little impact as buffer content is already copied
into message buffer which is places in proper memory for
optimized transfer.

Applied a little cleanup after nrfx 4.0 integration,

teburd commented on drivers/i2c/i2c_nrfx_twim_rtio.c:
```diff
@@ -188,6 +207,21 @@ static void event_handler(nrfx_twim_evt_t const *p_event, void *p_context)
+	int ret = 0;
+
+	if (p_event->xfer_desc.type == NRFX_TWIM_XFER_TX) {
+		ret = dmm_buffer_out_release(config->common.mem_reg,
```
Could this result in a large memcpy occuring in the ISR context? Won't this potentially disrupt time critical tasks?

At what size is the cost of a context swap worth considering perhaps to avoid holding up other tasks due to a memcpy?

Unlike the blocking driver which swaps back and forth to the calling thread, this does not. It expects you to be able to chain interupts together to avoid the context swap cost and reduce latency/overhead. But if you are doing large memcpy's in the ISR this isn't that beneficial. @bjarki-andreasen any thoughts?

teburd reviewed (commented):
Nothing blocking from me, but a note on the memcpy in ISR context which has me modestly concerned.

mstasiaknordic committed:
drivers: i2c: clean-up macros used in nrfx_twim

Removed unused macros. Moved common macros to common header.
Aligned naming and style to other nrf drivers.

Signed-off-by: Michał Stasiak <michal.stasiak@nordicsemi.no>

mstasiaknordic committed:
drivers: i2c: add DMM to i2c_nrfx_twim

Added DMM buffer preparation to I2C TWIM driver.
Adds little impact as buffer content is already copied
into message buffer which is places in proper memory for
optimized transfer.

Signed-off-by: Michał Stasiak <michal.stasiak@nordicsemi.no>
